comp/core/autodiscovery/listeners/snmp.go

```diff
@@ -54,14 +59,82 @@ const (
 	tagSeparator             = ","
 )
 
+// IPCounter interface defines operations for tracking IP authentication attempts
+type IPCounter interface {
+	Inc(ip string)
+	Dec(ip string)
+	Get(ip string) int
+	Set(ip string, count int)
+	Len() int
+	GetAll() map[string]int
+}
+
+type DeviceHashInfo struct {
```

Comment:
It doesn't look like we ever read Name, Description or IP.

I feel like we should be able to keep this more streamlined with a simple interface like:

```golang
interface {
    AddDevice(Name, Description, IP, BootTimeMs)
    Contains(Name, Description, IP, BootTimeMs)
}
```

which can be backed by either a simple list of struct (but then contains is a for-loop, likely ok but not optimal). Or a map with "basicHash" as keys, and a list of `{Name, Description, BootTimeMs, IP}` as the values